dotcom-rendering/src/model/insert-newsletter-signup.ts

```diff
@@ -0,0 +1,167 @@
+interface PlaceInArticle {
+	index: number;
+	distance: number;
+	afterText: boolean;
+	afterSupporting: boolean;
+}
+
+const isCloseAfterSupportingElement = (
+	index: number,
+	elements: CAPIElement[],
+): boolean => {
+	const elementIsSupporting = (element?: CAPIElement): boolean =>
+		!!(element && 'role' in element && element.role === 'supporting');
+
+	return (
+		elementIsSupporting(elements[index - 1]) ||
+		elementIsSupporting(elements[index - 2])
+	);
+};
+
+const isAfterText = (index: number, elements: CAPIElement[]): boolean =>
+	elements[index - 1]?._type ===
+	'model.dotcomrendering.pageElements.TextBlockElement';
+
+const getPlaces = (
+	targetIndex: number,
+	elements: CAPIElement[],
+): PlaceInArticle[] => {
+	return elements.map((element, index) => ({
+		index,
+		distance: Math.abs(index - targetIndex),
+		afterText: isAfterText(index, elements),
+		afterSupporting: isCloseAfterSupportingElement(index, elements),
```

Comment:
What about `afterSupporting` and `isNearSupporting`? I think that would make the code read nicer later on when checking if suitable